File: 0258_add_digits.swift
// https://leetcode.com/problems/add-digits/

class Solution {
	func addDigits(_ num: Int) -> Int {
		var num = num
		while num >= 10 {
			var temp = num
			var sum = 0
			while temp > 0 {
				sum += temp % 10
				temp = temp / 10
			}
			num = sum
		}
		return num
	}
}

File: 0258_add_digits.kt
// https://leetcode.com/problems/add-digits/description/

class Solution {
    fun addDigits(num: Int): Int {
        var n = num
        while (n >= 10) {
            var sum = 0
            while (n > 0) {
                sum += n % 10
                n /= 10
            }
            n = sum
        }
        return n
    }
}
